packages/versioned-playground/src/VersionedSquigglePlayground.tsx

```diff
@@ -1,106 +1,65 @@
 "use client";
-import { FC, ReactNode, useMemo, lazy, Suspense } from "react";
+import { FC, Suspense, lazy } from "react";
 
-import { useToast } from "@quri/ui";
+import { SquiggleVersion } from "./versions.js";
 
+/*
+ * We have to type all playground components explicitly; otherwise, TypeScript will complain with TS2742 "likely not portable" error.
+ * But we also need lazy imports and not load all playground versions immediately.
+ * This means that types should be imported immediately with `import { type ... }`, which are removed by TypeScript
+ * (`importsNotUsedAsValues`, https://www.typescriptlang.org/tsconfig#importsNotUsedAsValues, defaults to "remove"),
+ * but components themselves should be imported lazily.
+ */
 import {
-  SquiggleVersion,
-  checkSquiggleVersion,
-  defaultSquiggleVersion,
-} from "./versions.js";
+  SquigglePlaygroundProps_0_8_5,
+  SquigglePlaygroundProps_0_8_6,
+} from "./oldPlaygroundTypes.js";
+
+/*
+ * Please don't change the formatting of these imports and the following `playgroundByVersion` declaration unless you have to.
+ * It's edited with babel transformation in `publish-all.ts` script.
```

Comment:
So, if one were to manually change these, it would just be over-ridden? That sounds less like "please don't do this, it will be annoying for us", and more like, "if you do this, your work will be over-ridden."

Maybe change to,
"The following is autogenerated with .... . Don't edit it directly."

(Then, if we really need to, between the two of us, we could later change the policy)

Reply:
No, in almost all cases babel transformation will fail, so release GitHub action will fail. It won't be a silent overwrite.

It's not auto-generated, it's edited (surgically patched on AST level).